sqldelight-compiler/src/test/kotlin/com/squareup/sqldelight/core/queries/SelectQueryFunctionTest.kt

```diff
@@ -1320,6 +1322,32 @@ class SelectQueryFunctionTest {
     )
   }
 
+  @Test fun `type inference on boolean`() {
+    val file = FixtureCompiler.parseSql(
+      """
+      |CREATE TABLE exit (
+      |  wingsuit INTEGER AS Boolean NOT NULL
+      |);
+      |
+      |queryOne:
+      |SELECT * FROM exit WHERE wingsuit = :wingsuit AND :wingsuit = 1;
+      |
+      |queryTwo:
+      |SELECT * FROM exit WHERE :wingsuit = 1 AND wingsuit = :wingsuit;
+      """.trimMargin(),
+      tempFolder
+    )
+
+    val queryOne = file.namedQueries.first()
+    val generatorOne = SelectQueryGenerator(queryOne)
+    val queryTwo = file.namedQueries.first()
```

Comment:
> file.namedQueries.first()

I think there is a typo, it should be `.last()` or `[1]`
anyway, any ideas where the fix should be? is `MutableList<Argument>.findAndReplace` suitable for it?